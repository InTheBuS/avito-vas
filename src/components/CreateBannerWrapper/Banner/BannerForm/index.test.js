import React from "react"
import renderer from "react-test-renderer"

import {BannerForm} from "./index";

describe("Testing BannerForm component", () => {

    it("rendering when array has one object", () => {
        let testArrayWithOneElem = [{name: "body", value: "This is the body text", type: "text", label: "test label", placeholder: "test placeholder"},]
        const tree = renderer.create(<BannerForm fields={testArrayWithOneElem}/>).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it("rendering when array has multiple objects", () => {
        let testArrayWithMultipleElements = [
            {name: "body", value: "This is the body text", type: "text", label: "test label", placeholder: "test placeholder"},
            {name: "color", value: "#646464", type: "color", label: "test label", placeholder: "test placeholder"},
            {name: "image", value: "", type: "text", label: "test label", placeholder: "test placeholder"},
        ]
        const tree = renderer.create(<BannerForm fields={testArrayWithMultipleElements}/>).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it("rendering when array of inputs empty", () => {
        const tree = renderer.create(<BannerForm />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})