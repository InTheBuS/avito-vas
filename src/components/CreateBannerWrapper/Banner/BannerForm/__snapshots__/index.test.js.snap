// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing BannerForm component rendering when array has multiple objects 1`] = `
<div
  className="formWrapper"
/>
`;

exports[`Testing BannerForm component rendering when array has one object 1`] = `
<div
  className="formWrapper"
/>
`;

exports[`Testing BannerForm component rendering when array of inputs empty 1`] = `
<div
  className="formWrapper"
/>
`;
